tests/NoUnused/ExportsTest.elm

```diff
@@ -1549,3 +1545,101 @@ unused = 1
                           )
                         ]
         ]
+
+
+exposingAllTests : Test
+exposingAllTests =
+    describe "When exposing all"
+        [ test "works with a function" <|
+            \() ->
+                [ """
+module NotReported exposing (..)
+import Reported
+main = text ""
+"""
+                , """
+module Reported exposing (..)
+unused = ()
+"""
+                ]
+                    |> Review.Test.runOnModulesWithProjectData application rule
+                    |> Review.Test.expectErrorsForModules
+                        [ ( "Reported"
+                          , [ Review.Test.error
+                                { message = "Exposed function or value `unused` is never used outside this module."
+                                , details = unusedExposedElementDetails
+                                , under = ".."
+                                }
+                            ]
+                          )
+                        ]
+        , test "works with a type alias" <|
+            \() ->
+                [ """
+module NotReported exposing (..)
+import Reported
+main = text ""
+"""
+                , """
+module Reported exposing (..)
+type alias Unused = ()
+"""
+                ]
+                    |> Review.Test.runOnModulesWithProjectData application rule
+                    |> Review.Test.expectErrorsForModules
+                        [ ( "Reported"
+                          , [ Review.Test.error
+                                { message = "Exposed type or type alias `Unused` is never used outside this module."
+                                , details = unusedExposedElementDetails
+                                , under = ".."
+                                }
+                            ]
+                          )
+                        ]
+        , test "works with a custom type" <|
+            \() ->
+                [ """
+module NotReported exposing (..)
+import Reported
+main = text ""
+"""
+                , """
+module Reported exposing (..)
+type UnusedT = UnusedC
+"""
+                ]
+                    |> Review.Test.runOnModulesWithProjectData application rule
+                    |> Review.Test.expectErrorsForModules
+                        [ ( "Reported"
+                          , [ Review.Test.error
+                                { message = "Exposed type `UnusedT` is never used outside this module."
+                                , details = unusedExposedElementDetails
+                                , under = ".."
```

Comment:
`under` should point to `UnusedT` (Same for the other kinds of elements)

Reply:
I was wondering where the `Range` should point for these errors. Typically with an unused export, the error would point at the export itself, not the declaration. That's why I decided to point at the `..`. However that probably makes the error less useful since to fix it you have to go find the declaration anyway. I'll update the code to retain the `Range` from the declaration rather than moving it to the `..`.